packages/bull-vault/src/BullStrategy.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity =0.7.6;
+
+// contract
+import {ERC20} from "openzeppelin/token/ERC20/ERC20.sol";
+import {LeverageBull} from "./LeverageBull.sol";
+import {UniBull} from "./UniBull.sol";
+// lib
+import {StrategyMath} from "squeeth-monorepo/strategy/base/StrategyMath.sol"; // StrategyMath licensed under AGPL-3.0-only
+
+/**
+ * @notice BullStrategy contract
+ * @dev this is an abstracted BullStrategy in term of deposit and withdraw functionalities
+ * @author opyn team
+ */
+contract BullStrategy is ERC20, LeverageBull, UniBull {
+    using StrategyMath for uint256;
+
+    uint256 private constant ONE = 1e18;
+
+    address public immutable crab;
```

Comment:
We need to consider what if Crab changes and we want to set a new Crab contract address here, so probably not immutable.